### PR TITLE
styled-components fix: Define StyleSheet and change Node to React.ReactNode

### PR DESCRIPTION
**Pull request by nicogreenarry**
~As referenced [here](https://github.com/styled-components/styled-components/issues/1952), the `StyleSheet` and `Node` types were undefined. This PR defines `StyleSheet`, and replaces `Node` with `React.ReactNode`, following the example found [here](https://github.com/styled-components/styled-components/pull/1954).~
Update: The issue was that we needed a triple-slash directive referencing the "dom" library.

Note: There's a parallel [PR in the `styled-components` repo](https://github.com/styled-components/styled-components/pull/2019), since that repo still maintains typings.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
  - No tests added/changed; tests passed
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint styled-components` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - As referenced [here](https://github.com/styled-components/styled-components/issues/1952), the `StyleSheet` and `Node` types were undefined. This PR defines `StyleSheet`, and replaces `Node` with `React.ReactNode`, following the example found [here](https://github.com/styled-components/styled-components/pull/1954).
- [x] Increase the version number in the header if appropriate.
- ~[ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~

### CODE REVIEW

**Review by Igorbek — CHANGES_REQUESTED** [review]
I was not sure how this `target` prop was used and looked the code:

https://github.com/styled-components/styled-components/blob/cb7d90c3a138df0a5defc561574e9cb2e1f2bd81/src/models/StyleSheetManager.js#L27-L29

and

https://github.com/styled-components/styled-components/blob/cb7d90c3a138df0a5defc561574e9cb2e1f2bd81/src/models/StyleSheet.js#L49

it seems it is used as a DOM Node, not a React Node.
So the type was correct. Let me see what the issue was in the first place...

**Comment by nicogreenarry** [comment]
@Igorbek thanks for looking. `Node` isn't defined, though it's possible I have an incorrect typescript setup. I tried including "DOM" in the typescript `lib` config array, but `Node` was still undefined. Maybe it should be `HTMLElement`, or perhaps `HTMLElement | null`? I'm going to push that change, since `ReactNode` is definitely wrong.

**Comment by nicogreenarry** [comment]
Ah. I think what we actually want is to add this to the top of the file:
`/// <reference lib="dom" />`

**Comment by nicogreenarry** [comment]
Sigh. Maybe I'm seeing ghosts. The entire premise for this fix was that the `.d.ts` file failed to properly define/reference the DOM library. I've tried various fixes in this repo, and have run into various test failures. I tried fixing the issue again locally, on the premise that maybe these types are just fine as they are, and it's my own local typescript config that has problems. And, after upgrading to `styled-components@beta`, and ensuring that "dom" is listed as one of my lib's in my `tsconfig.json`, the compilation errors went away. So there's nothing to see here, folks, nothing at all!

File: types/styled-components/index.d.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
-// Type definitions for styled-components 3.0
+// Type definitions for styled-components 3.4
 // Project: https://github.com/styled-components/styled-components
 // Definitions by: Igor Oleinikov <https://github.com/Igorbek>
 //                 Ihor Chulinda <https://github.com/Igmat>
+//                 Nico Greenarry <https://github.com/NicoGreenarry>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 3.0
```
[the remark this answers is not in the document]
Comment (Igorbek) [reply types/styled-components/index.d.ts]: oops, didn't see you closed this.

**Pull request closed without merging**